### PR TITLE
feat(CB2-6212): Contingency Test Specialist Group 2

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## Contingency Testing - Specialist Tests Group 2 (COIF Annual & Seatbelt Installation Check)

_Added support for creation of test types 142, 146, 175, 177_
[CB2-6212](https://dvsa.atlassian.net/browse/CB2-6212)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+          children: [
+            {
+              name: 'testResult',
+              label: 'Result',
+              viewType: FormNodeViewTypes.HIDDEN,
+              editType: FormNodeEditTypes.RADIO,
+              options: [
+                { value: 'pass', label: 'Pass' },
+                { value: 'fail', label: 'Fail' },
+                { value: 'abandoned', label: 'Abandoned' }
```
Comment (gjulien-bjss) [inline src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts:52]:
```suggestion
                { value: 'fail', label: 'Fail' }
```
Now there is a button at the bottom of the page to abandon the test, the `abandon` option should be removed from the options

File: src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+              label: 'Additional details for abandoning',
+              editType: FormNodeEditTypes.TEXTAREA,
+              validators: [
+                {
+                  name: ValidatorNames.RequiredIfEquals,
+                  args: { sibling: 'testResult', value: 'abandoned' }
+                },
+                { name: ValidatorNames.MaxLength, args: 500 }
+              ]
+            },
```
Comment (gjulien-bjss) [inline src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts:99]: I _THINK_ the reasons for abandonning have been pulled into another tempalte/are now hidden.

```suggestion
```

**Review by kpearmanbjss — DISMISSED** [review]

File: src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+              label: 'Result',
+              viewType: FormNodeViewTypes.HIDDEN,
+              editType: FormNodeEditTypes.RADIO,
+              options: [
+                { value: 'pass', label: 'Pass' },
+                { value: 'fail', label: 'Fail' }
+              ],
+              validators: [
+                { name: ValidatorNames.HideIfNotEqual, args: { sibling: 'reasonForAbandoning', value: 'abandoned' } },
+                { name: ValidatorNames.HideIfNotEqual, args: { sibling: 'additionalCommentsForAbandon', value: 'abandoned' } },
```
Comment (Ruairidh-BJSS) [inline src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts:55]: I think you can remove the validation that involves the abandoned tests

**Review by Ruairidh-BJSS — CHANGES_REQUESTED** [review]

File: src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+              type: FormNodeTypes.CONTROL
+            },
+            {
+              name: 'testTypeName',
+              label: 'Description',
+              value: '',
+              disabled: true,
+
+              type: FormNodeTypes.CONTROL
+            },
```
Comment (Ruairidh-BJSS) [inline src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts:65]:
```suggestion
            },
            {
              name: 'reasonForAbandoning',
              type: FormNodeTypes.CONTROL,
              viewType: FormNodeViewTypes.HIDDEN,
              editType: FormNodeEditTypes.HIDDEN,
              value: null,
              required: true
            },
            {
              name: 'additionalCommentsForAbandon',
              type: FormNodeTypes.CONTROL,
              viewType: FormNodeViewTypes.HIDDEN,
              editType: FormNodeEditTypes.HIDDEN,
              value: null,
              required: true
            },
```

Comment (Ruairidh-BJSS) [reply src/app/forms/templates/test-records/section-templates/test/contingency-test-section-specialist-group2.template.ts:65]: This should be required but hidden

**Review by Ruairidh-BJSS — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**